src/main/rcsplit/rcsplit.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <ctype.h>
+
+#include <platform.h>
+
+#include "common/utils.h"
+
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+
+#include "fc/rc_controls.h"
+
+#include "io/beeper.h"
+#include "io/serial.h"
+
+#include "drivers/serial.h"
+
+#include "rcsplit/rcsplit.h"
+
+
+#define STATIC_ASSERT(condition, name) \
+    typedef char assert_failed_ ## name [(condition) ? 1 : -1 ]
+
+// This is public setting in the RunCam Split tab of CleanFlight Configurator
+// The user can bind the following mode with some aux channel in special range.
+static const box_t rcsplitBoxes[RCSPLIT_CHECKBOX_ITEM_COUNT] = {
+    { RCSPLIT_BOX_SIM_WIFI_BUTTON, "Wi-Fi Button", 0 },
+    { RCSPLIT_BOX_SIM_POWER_BUTTON, "Power Button", 1 },
+    { RCSPLIT_BOX_SIM_CHANGE_MODE, "Change Mode", 2 },
+};
+
+// mask of enabled RunCam Split Box IDs, calculated on startup based on enabled features. rcsplitBoxId_e is used as bit index
+uint32_t rcsplitModeActivationMask; // one bit per mode defined in rcsplitBoxId_e
+
+
+static uint32_t rcsplitActiveBoxIds;
+STATIC_ASSERT(sizeof(rcsplitActiveBoxIds) * 8 >= RCSPLIT_CHECKBOX_ITEM_COUNT, RCSPLIT_CHECKBOX_ITEMS_wont_fit_in_activeBoxIds);
+
+// the rcsplit mode setting in the EEPRom
+PG_REGISTER_ARRAY(modeActivationCondition_t, MAX_RC_SPLIT_MODE_ACTIVATION_CONDITION_COUNT, rcsplitModeActivationConditions, PG_RCSPLIT_MODE_ACTIVATION_PROFILE, 0);
+
+// communicate with camera device variables
+static serialPort_t *rcsplitSerialPort = NULL;
+static rcsplit_switch_state_t switchStates[RCSPLIT_CHECKBOX_ITEM_COUNT];
+static rcsplit_state_e cameraState = RCSPLIT_STATE_UNKNOWN;
+
+static unsigned char crc_high_first(unsigned char *ptr, unsigned char len)
```

Comment:
bigEndian is common term for high_first. Also CF uses camelCase. but simple `crc` should do as it is local function.

And CF uses uint8_t for 8bit data, `len` should be `int` or `unsigned` or `size_t`